Userland/Libraries/LibImageDecoderClient/Client.cpp

```diff
@@ -70,17 +67,26 @@ Optional<DecodedImage> Client::decode_image(ReadonlyBytes encoded_data)
     auto encoded_buffer = encoded_buffer_or_error.release_value();
 
     memcpy(encoded_buffer.data<void>(), encoded_data.data(), encoded_data.size());
-    auto response_or_error = try_decode_image(move(encoded_buffer));
+    auto response_or_error = try_decode_image(move(encoded_buffer), {});
 
     if (response_or_error.is_error()) {
-        dbgln("ImageDecoder died heroically");
-        return {};
+        if (mime_type.has_value()) {
+            return decode_image_with_known_mime_type(encoded_data, mime_type);
+        } else {
+            dbgln("ImageDecoder died heroically");
+            return {};
+        }
     }
 
     auto& response = response_or_error.value();
 
-    if (response.bitmaps().is_empty())
-        return {};
+    if (response.bitmaps().is_empty()) {
+        if (mime_type.has_value()) {
+            return decode_image_with_known_mime_type(encoded_data, mime_type);
+        } else {
+            return {};
+        }
+    }
```

Comment:
This seems redundant, and probably can be folded into a single call of `try_decode_image` with passing the optional mime type